include/sleipnir/autodiff/Gradient.hpp

```diff
@@ -57,11 +57,13 @@ class SLEIPNIR_DLLEXPORT Gradient {
   Variable m_variable;
   VectorXvar m_wrt;
 
+  std::vector<Expression*> m_graph;
+
   Eigen::SparseVector<double> m_g;
 
   Profiler m_profiler;
 
-  void CalculateImpl();
+  void Compute();
```

Comment:
This function needs a comment now that its purpose isn't self-evident as being "the internal implementation for Calculate()". What does the function do, and what are its preconditions and postconditions?